### DISCUSSION TITLE
Why do "colour.plotting.plot_sds_in_chromaticity_diagram_*" definitions produce unexpected chromaticity coordinates?

### DISCUSSION THREAD

**Discussion by ChrisABailey** [colour-science/colour#811] [Q&A] [answered]
I have been using the library to plot chromaticity diagrams for about a year and know just enough to get myself in trouble.

Typically I collect data from a spectroradiometer which I plot and I have not noticed any issues but today I loaded a new version of the library and now the points appear to be shifted on the CIE1931 and CIE1976 diagrams from what I measure and from what I calculate.  I put together the following to show my problem.   Unfortunately, I am not sure of my previous version so I am having a hard time finding anything in the changelog that indicates what may have happened.

The "x" markers in the diagrams are what I believe are the correct positions while the circles are the points generated internal to the function.

Note the example uses some internally created distributions but I first noticed the problem with measured data.

```
  def check_plot_SD(spectrums):
      colour_style()
      fig, ((ax1), (ax2)) = plt.subplots(1, 2)
      #plot from Spectral Distribution
      fig, c31ax = plot_sds_in_chromaticity_diagram_CIE1931(spectrums, standalone=False, axes=ax1)
      fig, c76ax = plot_sds_in_chromaticity_diagram_CIE1976UCS(spectrums, standalone=False, axes=ax2)
      
      for sd in spectrums:
          XYZ = colour.sd_to_XYZ(sd)
          xy = colour.XYZ_to_xy(XYZ/100)
          uv = xy_uv(xy) #this just converts from xy to u'v'
          print("%s xy = %s, u'v'  = %s" % (sd.name , str(xy), str(uv)))
          c31ax.plot(xy[0], xy[1], 'x-', color='black', ms=6)
          c76ax.plot(uv[0], uv[1], 'x-', color='black', ms=6)
      
  check_plot_SD([colour.sd_CIE_standard_illuminant_A(),colour.sd_CIE_illuminant_D_series([0.2,0.2]),colour.sd_CIE_illuminant_D_series([0.3,0.4])])

```

> CIE Standard Illuminant A xy = [ 0.44757152  0.4074404 ], u'v'  = [ 0.25596937  0.52429072]
>CIE xy (0.2, 0.2) - CIE Illuminant D Series xy = [ 0.19994293  0.19991918], u'v'  = [ 0.15998172  0.35991612]
>CIE xy (0.3, 0.4) - CIE Illuminant D Series xy = [ 0.30005264  0.39976999], u'v'  = [ 0.16676228  0.49991144]

![Figure 12](https://user-images.githubusercontent.com/4689838/116756903-fdbe3d80-a9da-11eb-923e-cd69a595e57d.png)

**Comment by thomasmansencal** [comment, marked as answer]
Hi @ChrisABailey,

This is a great question and it is pertaining to a change made to improve consistency across the plotting definitions made last year.

The cause for the shift is that previously, the illuminant used for plotting (and that could not be changed) was *CIE E*. It is fine and actually what you want when you plot radiation sources but it is causing issues when you plot reflectance or transmittance spectra. With that in mind, and for consistency with the other plotting definitions, I made D65 the default illuminant used for those definitions:
https://github.com/colour-science/colour/blob/d634b29c84a338f6abd5eb8423226febfc0a3774/colour/plotting/diagrams.py#L744

If you change your code to:

```python
      fig, c31ax = colour.plotting.plot_sds_in_chromaticity_diagram_CIE1931(spectrums, standalone=False, axes=ax1, plot_kwargs={'illuminant': colour.sd_ones()})
      fig, c76ax = colour.plotting.plot_sds_in_chromaticity_diagram_CIE1976UCS(spectrums, standalone=False, axes=ax2, plot_kwargs={'illuminant': colour.sd_ones()})
```

It should work:

![image](https://user-images.githubusercontent.com/99779/116761827-c39d7c00-aa6c-11eb-8c9c-8eef089226a9.png)

https://colab.research.google.com/drive/1F8bUn4xEF6nzta2HlF-mkgIhg1wx4Foy?usp=sharing

Cheers,

Thomas

**Reply by ChrisABailey** [reply]
Thanks, that works perfectly.
